src/devices/sound/sn76477.cpp

```diff
@@ -892,9 +892,11 @@ void sn76477_device::log_complete_state()
 
 void sn76477_device::open_wav_file()
 {
-	char wav_file_name[30];
+	std::string s = tag();
+	std::replace(s.begin(), s.end(), ':', '_');
+
+	char const* wav_file_name = util::string_format(LOG_WAV_FILE_NAME, s).c_str();
```

Comment:
This creates a reference-to-temporary – the string created by `util::string_format` will fall out of scope at the end of the statement, so `wav_file_name` will point to deallocated memory.  You can make `wav_file_name` a `std::string` (`LOG` can use `std::string directly with no need to call `.c_str()` and you can call `.c_str()` for other funtions that need a C string).

Reply:
My fault. I don't write C++ since years and I tought that .c_str() creates a new char* allocating the required space. Thx